app/uk/gov/hmrc/fileupload/microserviceGlobal.scala

```diff
@@ -17,83 +17,121 @@
 
-object MicroserviceAuthFilter extends AuthorisationFilter {
-  override lazy val authParamsConfig = AuthParamsControllerConfiguration
-  override lazy val authConnector = MicroserviceAuthConnector
+  var subscribe: (ActorRef, Class[_]) => Boolean = _
+  var publish: (AnyRef) => Unit = _
+  var withBasicAuth: BasicAuth = _
+  var eventStore: MongoEventStore = _
 
-  override def controllerNeedsAuth(controllerName: String): Boolean = ControllerConfiguration.paramsForController(controllerName).needsAuth
-}
+  implicit val reader = new UnitOfWorkReader(EventSerializer.toEventData)
+  implicit val writer = new UnitOfWorkWriter(EventSerializer.fromEventData)
 
-object MicroserviceGlobal extends DefaultMicroserviceGlobal with RunMode {
+  def basicAuthConfiguration(config: Configuration): BasicAuthConfiguration = {
+    def getUsers(config: Configuration): List[User] = {
+      config.getString("basicAuth.authorizedUsers").map { s =>
+        s.split(";").flatMap(
+          user => {
+            user.split(":") match {
+              case Array(username, password) => Some(User(username, password))
+              case _ => None
+            }
+          }
+        ).toList
+      }.getOrElse(List.empty)
+    }
 
-  override val auditConnector = MicroserviceAuditConnector
+    config.getBoolean("feature.basicAuthEnabled").getOrElse(false) match {
+      case true => BasicAuthEnabled(getUsers(config))
+      case false => BasicAuthDisabled
+    }
+  }
 
-  override def microserviceMetricsConfig(implicit app: Application): Option[Configuration] = app.configuration.getConfig(s"microservice.metrics")
+  lazy val database = new ReactiveMongoConnector(configuration, applicationLifecycle)
 
-  override val loggingFilter = MicroserviceLoggingFilter
+  lazy val db = database.mongoConnector.db
 
-  override val microserviceAuditFilter = MicroserviceAuditFilter
+  def onStart(): Unit = {
+    subscribe = actorSystem.eventStream.subscribe
+    publish = actorSystem.eventStream.publish
 
-  override val authFilter = None
+    // event store
+    if (play.Environment.simple().isProd && configuration.getBoolean("Prod.mongodb.replicaSetInUse").getOrElse(true)) {
+      eventStore = new MongoEventStore(db, writeConcern = commands.WriteConcern.ReplicaAcknowledged(n = 2, timeout = 5000, journaled = true))
+    } else {
+      eventStore = new MongoEventStore(db)
+    }
 
-  import play.api.libs.concurrent.Execution.Implicits._
+    withBasicAuth = BasicAuth(basicAuthConfiguration(configuration))
 
-  var subscribe: (ActorRef, Class[_]) => Boolean = _
-  var publish: (AnyRef) => Unit = _
+    // notifier
+    actorSystem.actorOf(NotifierActor.props(subscribe, find, sendNotification), "notifierActor")
+    actorSystem.actorOf(StatsActor.props(subscribe, find, sendNotification, saveFileQuarantinedStat,
+      deleteVirusDetectedStat, deleteFileStoredStat), "statsActor")
 
-  var withBasicAuth: BasicAuth = _
+  }
 
-  lazy val db = DefaultMongoConnection.db
+  onStart()
```

Comment:
I am not sure if this is needed anymore. In 2.3 configuration was available quite late, thats why we introduced this mutable vars. I think we don't need that anymore with 2.5